map/search_api.cpp

```diff
@@ -0,0 +1,295 @@
+}
+
+bool SearchAPI::SearchInDownloader(storage::DownloaderSearchParams const & params)
+{
+  SearchParams p;
+  p.m_query = params.m_query;
+  p.m_inputLocale = params.m_inputLocale;
+  p.m_position = m_delegate.GetCurrentPosition();
+  SetViewportIfPossible(p);  // Search request will be delayed if viewport is not available.
+  p.m_maxNumResults = SearchParams::kDefaultNumResultsEverywhere;
+  p.m_mode = Mode::Downloader;
+  p.m_suggestsEnabled = false;
+  p.m_needAddress = false;
+  p.m_needHighlighting = false;
+
+  p.m_onResults = DownloaderSearchCallback(static_cast<DownloaderSearchCallback::Delegate &>(*this),
+                                           m_index, m_infoGetter, m_storage, params);
+
+  return Search(p, true /* forceSearch */);
+}
+
+void SearchAPI::PokeSearchInViewport(bool forceSearch)
+{
+  if (!m_isViewportInitialized || !IsViewportSearchActive())
+    return;
+
+  // Copy is intentional here, to skip possible duplicating requests.
+  auto params = m_searchIntents[static_cast<size_t>(Mode::Viewport)].m_params;
+  SetViewportIfPossible(params);
+  params.m_position = m_delegate.GetCurrentPosition();
+  Search(params, forceSearch);
+}
+
+void SearchAPI::CancelSearch(Mode mode)
+{
+  ASSERT_NOT_EQUAL(mode, Mode::Count, ());
+
+  if (mode == Mode::Viewport)
+  {
+    // TODO: delete me after Cian project is finished.
+    m_cianSearchMode = false;
+
+    m_delegate.ClearViewportSearchResults();
+    m_delegate.SetSearchDisplacementModeEnabled(false /* enabled */);
+  }
+
+  auto & intent = m_searchIntents[static_cast<size_t>(mode)];
+  intent.m_params.Clear();
+  CancelQuery(intent.m_handle);
+}
+
+void SearchAPI::CancelAllSearches()
+{
+  for (size_t i = 0; i < static_cast<size_t>(Mode::Count); ++i)
+    CancelSearch(static_cast<Mode>(i));
+}
+
+void SearchAPI::RunUITask(std::function<void()> fn) { return m_delegate.RunUITask(fn); }
+
+void SearchAPI::SetHotelDisplacementMode()
+{
+  return m_delegate.SetSearchDisplacementModeEnabled(true);
+}
+
+bool SearchAPI::IsViewportSearchActive() const
+{
+  return !m_searchIntents[static_cast<size_t>(Mode::Viewport)].m_params.m_query.empty();
+}
+
+void SearchAPI::ShowViewportSearchResults(Results const & results)
+{
+  return m_delegate.ShowViewportSearchResults(results);
+}
+
+void SearchAPI::ClearViewportSearchResults() { return m_delegate.ClearViewportSearchResults(); }
+
+bool SearchAPI::IsLocalAdsCustomer(Result const & result) const
+{
+  return m_delegate.IsLocalAdsCustomer(result);
+}
+
+bool SearchAPI::Search(SearchParams const & params, bool forceSearch)
+{
+  if (m_delegate.ParseMagicSearchQuery(params))
+    return false;
+
+  auto const mode = params.m_mode;
+  auto & intent = m_searchIntents[static_cast<size_t>(mode)];
+
+#ifdef FIXED_LOCATION
+  SearchParams currParams = params;
+  if (currParams.IsValidPosition())
+  {
+    // TODO (@y): fix this
+    m_fixedPos.GetLat(currParams.m_lat);
+    m_fixedPos.GetLon(rParams.m_lon);
```

Comment:
s/rParams/currParams/

Reply:
We agreed to remove FIXED_LOCATION blocks.